#!/usr/bin/env python
PACKAGE = "sl_crazyflie_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("KP_x",   double_t,    0, "An Integer parameter", 3.3,  0, 10)
gen.add("KI_x",    double_t,    0, "An Integer parameter", 0.0,  0, 2)
gen.add("KD_x",    double_t,    0, "An Integer parameter", 0.15,  0, 2)
gen.add("max_x_error",    double_t,    0, "An Integer parameter", 1.5,  0, 4)
gen.add("max_x_d_filter",    int_t,    0, "An Integer parameter", 6,  0, 20)

gen.add("KP_y",   double_t,    0, "An Integer parameter", 3.3,  0, 10)
gen.add("KI_y",    double_t,    0, "An Integer parameter", 0.0,  0, 2)
gen.add("KD_y",    double_t,    0, "An Integer parameter", 0.15,  0, 2)
gen.add("max_y_error",    double_t,    0, "An Integer parameter", 1.5,  0, 4)
gen.add("max_y_d_filter",    int_t,    0, "An Integer parameter", 6,  0, 20)

gen.add("KP_z",   double_t,    0, "An Integer parameter", 3.0,  0, 4)
gen.add("KI_z",    double_t,    0, "An Integer parameter", 0,  0, 20)
gen.add("KD_z",    double_t,    0, "An Integer parameter", 0.6,  0, 20)
gen.add("max_z_error",    double_t,    0, "An Integer parameter", 2.,  0, 4)
gen.add("max_z_d_filter",    int_t,    0, "An Integer parameter", 6,  0, 20)

gen.add("KP_yaw",   double_t,    0, "An Integer parameter", 1.0,  0, 10.0)
gen.add("KI_yaw",    double_t,    0, "An Integer parameter", 0.0,  0, 2)
gen.add("KD_yaw",    double_t,    0, "An Integer parameter", 0.0,  0, 2)
gen.add("max_yaw_error",    double_t,    0, "An Integer parameter", 1.7,  0, 4)
gen.add("max_yaw_d_filter",    int_t,    0, "An Integer parameter", 1,  0, 20)

exit(gen.generate(PACKAGE, "sl_crazyflie_controller", "pid_position"))